docs/rfcs/0003-rfc-controller_refactor.md

```diff
@@ -0,0 +1,99 @@
+- Status: In review
+- Date: 2023-06-20
+- Author: [@gela](https://github.com/gela)
+- RFC PR: https://github.com/mozilla-mobile/mozilla-vpn-client/pull/7269
+- Implementation GitHub issue: TBD.
+
+
+## Summary
+Introduce a new `ConnectionManager` component which sits between the frontend and the controller to handle all connectivity related logic and keep the controller code lean and minimal.
```

Comment:
nit:
```suggestion
Proposes a new `ConnectionManager` component which sits between the frontend and the controller to handle all connectivity related logic and keep the controller code lean and minimal.
```

Reply:
"code lean and minimal". Can we be more concrete here? What does "lean and minimal" mean?